gluon-mesh-vpn-wireguard/files/lib/gluon/gluon-mesh-wireguard/checkuplink

```diff
@@ -52,63 +59,36 @@ if [ "$(uci get wireguard.wireguard.enabled)" == "true" ] || [ "$(uci get wiregu
-        	until [ $I -eq $ANZAHL ]                                                                           
-        	do                                                                                                 
-                	I=`expr $I + 1`                                                                            
-                	#Von allen Servern die durschnittliche potenzielle Geschwindigkeit pro Wireguard-Verbindung
-                	URL=$(uci get wireguard.peer_$I.endpoint | cut -d':' -f1)                        
-			SPEED2=$(curl https://$URL/speed.php -k)                                           
-                	if [ $SPEED2 -gt $SPEED1 ]              # Wenn aktueller Server hoehren Wert hat,
-                        		then AUSWAHL=$I                 # Servernummer speichern                 
-                        	SPEED1=$SPEED2                                                           
-                	fi                                                                               
-        	done                                                                                     
-        	URL=$(uci get wireguard.peer_$AUSWAHL.endpoint | cut -d':' -f1)                          
+		modprobe wireguard                                                       
+
+                NUMBER_OF_PEERS=$(uci show wireguard | egrep -e peer_[0-9].endpoint | wc -l)
+                PEER=$(awk 'BEGIN{srand();print int(rand()*($NUMBER_OF_PEERS)+1) }')
                                                                                                  
 		#Die alten peers loeschen
 		wg set $INTERFACE peer  $(wg |  awk 'BEGIN {RS=""} /endpoint/ {print $2}') remove
-                batctl if del gre
-                ip link delete dev gre
+                batctl if del vxlan-wan
+                ip link delete dev vxlan-wan
                 ip link delete dev $INTERFACE
                 PUBLICKEY=$(uci get wireguard.wireguard.privatekey | wg pubkey)
                 echo $(uci get wireguard.wireguard.privatekey) > /tmp/wgpriv
 
                 # Public Key zum vorher ausgesuchten Server hochladen
-                IP=$(curl -s -k --data-urlencode "pubkey=$PUBLICKEY" https://$URL/wireguard.php)
+                #curl -s -k --data-urlencode "pubkey=$PUBLICKEY" https://$URL/wireguard.php
 		ip link add dev $INTERFACE type wireguard
-                ip address add dev $INTERFACE $IP/16
                 wg set $INTERFACE private-key /tmp/wgpriv
                 sleep 5
 
-                wg set $INTERFACE peer $(uci get wireguard.peer_$AUSWAHL.publickey) persistent-keepalive 25 allowed-ips $(uci get wireguard.wireguard.gretapip)/32 endpoint $(uci get wireguard.peer_$AUSWAHL.endpoint)
+                wg set $INTERFACE peer $(uci get wireguard.peer_$PEER.publickey) persistent-keepalive 25 allowed-ips fe80::/10,$(uci get wireguard.wireguard.mcastgroup)/128 endpoint $(uci get wireguard.peer_$PEER.endpoint)
```

Comment:
How does that work on the gateway side? Every node accept fe80::/10 from the gateway, what would Gateways set as AllowedIPs?

Reply:
The gateway also accepts all fe80 just to make sure we can then do vxlan on the discovered addresses.